src/main/java/graphql/cachecontrol/CacheControl.java

```diff
@@ -0,0 +1,195 @@
+     *
+     * @return this object builder style
+     */
+    public CacheControl hint(ExecutionPath path, Integer maxAge) {
+        return hint(path, maxAge, Scope.PUBLIC);
+    }
+
+    /**
+     * This creates a cache control hint for the specified field being fetched
+     *
+     * @param dataFetchingEnvironment the path to the field that has the cache control hint
+     * @param maxAge                  the caching time in seconds
+     * @param scope                   the scope of the cache control hint
+     *
+     * @return this object builder style
+     */
+    public CacheControl hint(DataFetchingEnvironment dataFetchingEnvironment, Integer maxAge, Scope scope) {
+        assertNotNull(dataFetchingEnvironment);
+        assertNotNull(scope);
+        hint(dataFetchingEnvironment.getExecutionStepInfo().getPath(), maxAge, scope);
+        return this;
+    }
+
+    /**
+     * This creates a cache control hint for the specified field being fetched with a PUBLIC scope
+     *
+     * @param dataFetchingEnvironment the path to the field that has the cache control hint
+     * @param maxAge                  the caching time in seconds
+     *
+     * @return this object builder style
+     */
+    public CacheControl hint(DataFetchingEnvironment dataFetchingEnvironment, Integer maxAge) {
+        hint(dataFetchingEnvironment, maxAge, Scope.PUBLIC);
+        return this;
+    }
+
+    /**
+     * This creates a cache control hint for the specified field being fetched with a specified scope
+     *
+     * @param dataFetchingEnvironment the path to the field that has the cache control hint
+     * @param scope                   the scope of the cache control hint
+     *
+     * @return this object builder style
+     */
+    public CacheControl hint(DataFetchingEnvironment dataFetchingEnvironment, Scope scope) {
+        return hint(dataFetchingEnvironment, null, scope);
+    }
+
+    /**
+     * Creates a new CacheControl object that can be used to trick caching hints
+     *
+     * @return the new object
+     */
+    public static CacheControl newCacheControl() {
+        return new CacheControl();
+    }
+
+    /**
+     * This will record the values in the cache control object into the provided execution result object which creates a new {@link graphql.ExecutionResult}
+     * object back out
+     *
+     * @param executionResult the starting execution result object
+     *
+     * @return a new execution result with the hints in the extensions map.
+     */
+    public ExecutionResult addTo(ExecutionResult executionResult) {
+        assertTrue(executionResult instanceof ExecutionResultImpl, "You must pass in an ExecutionResult based on graphql.ExecutionResultImpl");
+        Map<Object, Object> currentExtensions = executionResult.getExtensions();
+        if (currentExtensions == null) {
+            currentExtensions = Collections.emptyMap();
+        }
+        currentExtensions = new LinkedHashMap<>(currentExtensions);
```

Comment:
I would move this into putHintsInExtensionsMap and return it like:
   private Map<> putHintsInExtensionsMap(Map<Object, Object> extensions) {
        Map<> currentExtensions = new LinkedHashMap<>(extensions)
         List<Map<String, Object>> recordedHints = map(hints, Hint::toMap);

         Map<String, Object> cacheControl = new LinkedHashMap<>();
        cacheControl.put("version", 1);
        cacheControl.put("hints", recordedHints);

         currentExtensions.put("cacheControl", cacheControl);
        return currentExtensions;
    }